### ISSUE TITLE
[bitnami/elasticsearch]: Use x-pack basic image instead of OSS version in the chart

### ISSUE THREAD

**Issue by nikhilno1** [bitnami/charts#2957]
I understand the bitnami elasticsearch image uses the OSS version. 
I wanted to know if it is possible to use the x-pack basic image instead of the OSS from these helm charts?
Thanks.

**Comment by dani8art** [comment]
Hi @nikhilno1 we implemented this chart using OSS version of elasticsearch and taking the advantages of the configuration logic that is implemented [in the container](https://github.com/bitnami/bitnami-docker-elasticsearch/blob/master/7/debian-10/rootfs/opt/bitnami/scripts/libelasticsearch.sh), regarding that, you can try to use the x-pack version of elasticsearch but unfortunately we can not guarantee it will work with our chart. 

BTW, maybe we could help giving you an OSS solution for the problem, even you can maybe contribute adding a new feature to our chart solving your problem with an OSS. What do you want to aim using this image?

**Comment by nikhilno1** [comment]
Thank you for the response. I am using it to build a log analytics solution. I have an on-prem version (that uses x-pack basic) and want to move it to cloud. I would like to get as many features out of the box so that I have fewer customizations to deal with.
For e.g. x-pack basic can give me monitoring, RBAC, index management and bunch of other things that I can use straight-away.
Is not supporting x-pack basic image a conscious decision to stay clear of the licensing issue or just that nobody has done it before?

**Comment by nikhilno1** [comment]
Hi @dani8art, if I have to change the ES image from OSS to Basic, what changes are needed? Would you be able to give some pointers? Thanks.

**Comment by dani8art** [comment]
Hi @nikhilno1 sorry for my late response,

> Is not supporting x-pack basic image a conscious decision to stay clear of the licensing issue or just that nobody has done it before?

Yes it is a matter of licensing, our open source catalog only ships open source licensed software in containers and charts.

>  if I have to change the ES image from OSS to Basic, what changes are needed? 

Regarding the second question, I don't know what you would need to add, what I can say is what we do in our image.

- In the repository for `bitnami/elasticseach` you could see all the bash configuration logic that we apply to configure the deployment https://github.com/bitnami/bitnami-docker-elasticsearch

Basically the image has:
- `entrypoint.sh` that is the entry point for the container, here we call to the `setup.sh` and then execute the command given through docker run, by default it is `run.sh`
- `setup.sh` in this script we validate user parameters given in env vars and configure elasticsearch using `libelasticsearch.sh`
- `run.sh` execute the java command with some additional options.

**Comment by nikhilno1** [comment]
Thank you @dani8art for the detailed and clear response. Appreciate it.
